sed/io.py

```diff
@@ -75,89 +113,97 @@ def res_to_xarray(res, bin_names, bin_axes, metadata=None):
 
     for name in bin_names:
         try:
-            xres[name].attrs['unit'] = DEFAULT_UNITS[name]
+            xres[name].attrs["unit"] = DEFAULT_UNITS[name]
         except KeyError:
             pass
 
-    xres.attrs['units'] = 'counts'
-    xres.attrs['long_name'] = 'photoelectron counts'
+    xres.attrs["units"] = "counts"
+    xres.attrs["long_name"] = "photoelectron counts"
 
     if metadata is not None:
-        xres.attrs['metadata'] = metadata
+        xres.attrs["metadata"] = metadata
 
     return xres
 
-def xarray_to_h5(data, faddr, mode='w'):
-    """ Save xarray formatted data to hdf5
+
+def xarray_to_h5(data: xr.DataArray, faddr: str, mode: str = "w"):
+    """Save xarray formatted data to hdf5
+
     Args:
-        data (xarray.DataArray): input data
+        data: input data
         faddr (str): complete file name (including path)
         mode (str): hdf5 read/write mode
+
+    Raises:
+        Warning: subfunction warns if elements have been converted into strings for
+        saving.
+
     Returns:
     """
     with h5py.File(faddr, mode) as h5File:
 
-        print(f'saving data to {faddr}')
-
-        # Saving data
+        print(f"saving data to {faddr}")
 
-        ff = h5File.create_group('binned')
-
-        # make a single dataset
-        ff.create_dataset('BinnedData', data=data.data)
+        # Saving data, make a single dataset
+        h5File.create_dataset("binned/BinnedData", data=data.data)
 
         # Saving axes
-        aa = h5File.create_group("axes")
-        ax_n = 0
+        axesGroup = h5File.create_group("axes")
```

Comment:
camel_case is prefered for naming variables

Reply:
Variables in binning.py are mostly using camelCase. function names are using snake_case. Should we stick to this, then?

Reply:
I apologized by this confussion. snake_case is the prefered way of writing variables. CamelCase is used "only" in class names (https://pythonguides.com/python-naming-conventions/)

Reply:
I read this that both are fine for variable names. If you want to change this now, we need to change it everywhere, also in binning.py, etc. I am not convinced that this is necessary. Maybe something to discuss in a meeting soon

Reply:
You're absolutely right and I'll create a PR.
My point with this, as discussed with @steinnymir, is that for an external user of the library, if she/he sees inconsistent use of snake/camel case, docstrings with different styles, extensive tests, linting applied randomly, etc., it tells a bit about the care that the developers put into the library.